View and edit browser profile

SuaYoo opened the pull request:
(https://github.com/webrecorder/browsertrix-cloud/issues/199#issuecomment-1102919825) List visited URLs in side bar and refactor profile creation flow.

### Manual testing

### Screenshots
Prerequisite: Chromium-based browser
1. Go to Browser Profiles and start creating a new profile with "New Browser Profile"
2. Interact with browser, visiting different URLs. Verify that URLs besides starting URL shows under "Newly Visited"
3. Click "Next" and save. Verify new profile is created as expected
4. Click "Start Viewer". Verify profile loads as expected
5. Click different origins under "Visited URLs" in sidebar. Verify browser loads URL (may take a few seconds)
6. Navigate to new URLs and click "Save Changes". Refresh to verify changes are saved
7. Click "Duplicate Browser Profile" and complete workflow. Verify profile is created as expected

### Screenshots
**Browser profile detail view:**
<img width="1023" alt="Screen Shot 2022-04-19 at 5 07 50 PM" src="https://user-images.githubusercontent.com/4672952/164121005-ac69f138-d74d-4fd0-8e47-b6b937dcd64e.png">

**Detail view with profile loaded:**
<img width="1018" alt="Screen Shot 2022-04-19 at 5 09 08 PM" src="https://user-images.githubusercontent.com/4672952/164121194-e2ad6969-2359-4093-aa7a-67401781cdad.png">

**New browser profile view:**
<img width="1017" alt="Screen Shot 2022-04-19 at 5 03 32 PM" src="https://user-images.githubusercontent.com/4672952/164120983-4ae275d5-4956-4153-b35c-ded7b02fab50.png">

**Duplicating browser profile:**
<img width="1027" alt="Screen Shot 2022-04-19 at 5 10 46 PM" src="https://user-images.githubusercontent.com/4672952/164121227-5ea883ee-95d7-434f-9b54-2183108dda60.png">

SuaYoo committed:
update comment

SuaYoo committed:
update verb

SuaYoo committed:
replace dialog with direct preview

SuaYoo committed:
navigate on launch

SuaYoo committed:
get origins from ping

SuaYoo committed:
move browser to component

SuaYoo committed:
add section to detail page

SuaYoo committed:
enable fullscreen

SuaYoo committed:
embed profile in detail page

SuaYoo committed:
move origins to profile browser component

SuaYoo committed:
update new dialog

SuaYoo committed:
update extending ui

SuaYoo committed:
adjust layout

SuaYoo committed:
remove profile id option

SuaYoo committed:
update copy name

SuaYoo committed:
update language

SuaYoo committed:
fix initial name

ikreymer commented:
Looks good and everything seems to work! Was thinking of some minor tweaks:
- Can the iframe be hidden as soon as save is clicked, especially in editing mode as there's no dialog. When saving, it stops the browser, which then shows the devtools disconnect message and user might think something went wrong?
- Wonder if ping interval should be made more frequent to get newly visited sites?
- The site lists origins not exact URLs, so perhaps should be labeled 'Visited Sites'? or Profile contains sites?
The main idea is to indicate which origins might have cookie/session storage data associated with them as part of the profile, not a listing of URLs (and perhaps to avoid suggesting that the URLs are archived)

SuaYoo committed:
decrease poll interval

SuaYoo committed:
hide profile after editing

SuaYoo committed:
adjust save ux

SuaYoo committed:
update language

SuaYoo committed:
fix overlay

SuaYoo committed:
save updated profile

SuaYoo committed:
disable ping when browser closes

SuaYoo commented:
@ikreymer pushed suggested changes

> Can the iframe be hidden as soon as save is clicked, especially in editing mode as there's no dialog. When saving, it stops the browser, which then shows the devtools disconnect message and user might think something went wrong?

Added a loading overlay after clicking the save button and reset the iframe. I changed "Save Changes" to "Done Editing" and added a confirmation message to make it clearer that saving will reset the iframe:
<img width="472" alt="Screen Shot 2022-04-20 at 6 23 05 PM" src="https://user-images.githubusercontent.com/4672952/164353916-dc0b395d-1166-42f8-b27e-93c200d721ce.png">

> Wonder if ping interval should be made more frequent to get newly visited sites?

Updated to ping every 2 seconds, feels much more responsive and API is holding up!

> The site lists origins not exact URLs, so perhaps should be labeled 'Visited Sites'? or Profile contains sites?

Updated to "Visited Sites" and "New Sites" with info tooltips:
<img width="329" alt="Screen Shot 2022-04-20 at 6 22 15 PM" src="https://user-images.githubusercontent.com/4672952/164354129-efe3ea5b-9632-47e1-b170-fa1e00f6f9d3.png">
<img width="379" alt="Screen Shot 2022-04-20 at 6 22 55 PM" src="https://user-images.githubusercontent.com/4672952/164354131-d54eb262-c566-424b-a23c-1779bc3711ff.png">

SuaYoo committed:
fix browser compatibility message

SuaYoo committed:
adjust copy

SuaYoo committed:
Edit browser profile name and description (#221)

SuaYoo committed:
link to crawl templates

SuaYoo commented:
Associated crawl templates linked in detail:
<img width="287" alt="Screen Shot 2022-04-20 at 8 46 54 PM" src="https://user-images.githubusercontent.com/4672952/164368132-8baf8190-b043-4e62-b5aa-6725c07b6675.png">

SuaYoo committed:
duplicate from list

SuaYoo commented:
Menu item to duplicate profile:
<img width="1160" alt="Screen Shot 2022-04-20 at 9 18 29 PM" src="https://user-images.githubusercontent.com/4672952/164371257-fa10c417-c8a1-488d-a1af-08966d1661f2.png">
